x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elser/ElserMlNodeService.java

```diff
@@ -229,6 +234,25 @@ public void infer(Model model, List<String> input, Map<String, Object> taskSetti
         );
     }
 
+    @Override
+    public void putModel(Model model, ActionListener<Boolean> listener) {
+        if (model instanceof ElserMlNodeModel == false) {
+            listener.onFailure(
+                new IllegalStateException("Error starting model, [" + model.getConfigurations().getModelId() + "] is not an elser model")
+            );
+            return;
+        } else {
+            String modelVariant = ((ElserMlNodeModel) model).getServiceSettings().getModelVariant();
+            var fieldNames = List.<String>of();
+            var input = new TrainedModelInput(fieldNames);
+            var config = TrainedModelConfig.builder().setInput(input).setModelId(modelVariant).build();
+            PutTrainedModelAction.Request putRequest = new PutTrainedModelAction.Request(config, false, true);
+            executeAsyncWithOrigin(client, ML_ORIGIN, PutTrainedModelAction.INSTANCE, putRequest, listener.delegateFailure((l, r) -> {
```

Comment:
Hmm I wonder if we should be using a new origin for inference instead of `ML_ORIGIN` 🤔 @davidkyle ? Or do we need this as `ML_ORIGIN` so it gets the permissions correct?

Reply:
I didn't really consider using a different origin. I'll have to look into what the impact of the origin is. Thanks for pointing this out!

Reply:
Looks like there is already an inference origin, so I switched to using that. Thanks @jonathan-buttner !